LinkageCheckerRuleTest fails in Java 11

# testExecute_shouldFilterExclusionRule

```
2020-12-02T21:51:10.3495073Z Argument(s) are different! Wanted:
2020-12-02T21:51:10.3495698Z log.error(
2020-12-02T21:51:10.3496372Z     startsWith("Linkage Checker rule found 93 errors:")
2020-12-02T21:51:10.3496981Z );
2020-12-02T21:51:10.3499523Z -> at com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRuleTest.testExecute_shouldFilterExclusionRule(LinkageCheckerRuleTest.java:463)
2020-12-02T21:51:10.3501983Z Actual invocations have different arguments:
2020-12-02T21:51:10.3502611Z log.error(
2020-12-02T21:51:10.3503210Z     "Linkage Checker rule found 98 errors:
2020-12-02T21:51:10.3503938Z Class javax.inject.Provider is not found;
2020-12-02T21:51:10.3504659Z   referenced by 21 class files
```

https://pipelines.actions.githubusercontent.com/Fu5O9i8FcLso5jbRtRss9DmEb7w3qrbqwwoyZic6OgsmkRboUU/_apis/pipelines/1/runs/60/signedlogcontent/3?urlExpires=2020-12-03T15%3A10%3A10.5222231Z&urlSigningMethod=HMACV1&urlSignature=4em%2ByToSVMTFsdPY2cEScMRbd0DyuVr4BFwtukrAmH0%3D


# testExecute_shouldFailForBadProject

```
[ERROR]   LinkageCheckerRuleTest.testExecute_shouldFailForBadProject:237 expected to start with:
    Linkage Checker rule found 112 errors:
but was:
    Linkage Checker rule found 117 errors:
    Class com.google.net.rpc3.client.RpcStubDescriptor is not found;
```

# testExecute_shouldFailForBadProjectWithBundlePackaging

```
[ERROR] testExecute_shouldFailForBadProjectWithBundlePackaging(com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRuleTest)  Time elapsed: 3.398 s  <<< FAILURE!
org.mockito.exceptions.verification.junit.ArgumentsAreDifferent: 

Argument(s) are different! Wanted:
log.error(
    startsWith("Linkage Checker rule found 112 errors:")
);
-> at com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRuleTest.testExecute_shouldFailForBadProjectWithBundlePackaging(LinkageCheckerRuleTest.java:441)
Actual invocations have different arguments:
log.error(
    "Linkage Checker rule found 117 errors:
Class com.google.net.rpc3.client.RpcStubDescriptor is not found;
  referenced by 21 class files
```

# Classes in javax.activation

The new 5 linkage errors are from `javax.activation` classes:

```
2020-12-02T21:51:10.5376880Z Class javax.activation.FileDataSource is not found;
2020-12-02T21:51:10.5377536Z   referenced by 1 class file
2020-12-02T21:51:10.5378642Z     javax.mail.internet.MimeBodyPart (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5379444Z   Cause:
2020-12-02T21:51:10.5379748Z     Unknown
2020-12-02T21:51:10.5380285Z Class javax.activation.DataHandler is not found;
2020-12-02T21:51:10.5380898Z   referenced by 4 class files
2020-12-02T21:51:10.5381950Z     javax.mail.internet.MimeBodyPart (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5383427Z     javax.mail.internet.MimeMessage (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5384877Z     javax.mail.internet.MimeUtility (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5386540Z     javax.mail.internet.PreencodedMimeBodyPart (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5387514Z   Cause:
2020-12-02T21:51:10.5387805Z     Unknown
2020-12-02T21:51:10.5388497Z Class javax.activation.ActivationDataFlavor is not found;
2020-12-02T21:51:10.5389232Z   referenced by 5 class files
2020-12-02T21:51:10.5390513Z     org.apache.geronimo.mail.handlers.HtmlHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5392495Z     org.apache.geronimo.mail.handlers.MessageHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5394592Z     org.apache.geronimo.mail.handlers.MultipartHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5396711Z     org.apache.geronimo.mail.handlers.TextHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5398646Z     org.apache.geronimo.mail.handlers.XMLHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5399639Z   Cause:
2020-12-02T21:51:10.5399948Z     Unknown
2020-12-02T21:51:10.5400595Z Class javax.activation.DataContentHandler is not found;
2020-12-02T21:51:10.5401295Z   referenced by 3 class files
2020-12-02T21:51:10.5402637Z     org.apache.geronimo.mail.handlers.MessageHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5404715Z     org.apache.geronimo.mail.handlers.MultipartHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5406745Z     org.apache.geronimo.mail.handlers.TextHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.5407748Z   Cause:
2020-12-02T21:51:10.5408054Z     Unknown
...
2020-12-02T21:51:10.6840121Z Class javax.activation.DataSource is not found;
2020-12-02T21:51:10.6840705Z   referenced by 9 class files
2020-12-02T21:51:10.6842064Z     com.google.appengine.api.utils.HttpRequestParser (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6844263Z     com.google.apphosting.utils.servlet.MultipartMimeUtils (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6846151Z     javax.mail.MultipartDataSource (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6847627Z     javax.mail.internet.MimeMultipart (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6849223Z     javax.mail.internet.MimePartDataSource (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6850986Z     javax.mail.internet.MimeUtility (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6852500Z     javax.mail.util.ByteArrayDataSource (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6854336Z     org.apache.geronimo.mail.handlers.MessageHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6856309Z     org.apache.geronimo.mail.handlers.TextHandler (com.google.appengine:appengine-api-1.0-sdk:1.9.64)
2020-12-02T21:51:10.6857328Z   Cause:
2020-12-02T21:51:10.6857622Z     Unknown
```

--- a/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
+++ b/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
@@ -234,7 +234,14 @@ public class LinkageCheckerRuleTest {
       verify(mockLog, times(1)).error(errorMessageCaptor.capture());
 
       String errorMessage = errorMessageCaptor.getValue();
-      Truth.assertThat(errorMessage).startsWith("Linkage Checker rule found 112 errors:");
+
+      // Java 11 removed javax.activation package. Therefore the number of expected error differs
+      // between Java 8 and Java 11.
+      // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1856
+      int expectedErrorCount = System.getProperty("java.version").startsWith("1.8.") ? 112 : 117;
+
+      Truth.assertThat(errorMessage)
+          .startsWith("Linkage Checker rule found " + expectedErrorCount + " errors:");
       Truth.assertThat(errorMessage)
           .contains(
               "Problematic artifacts in the dependency tree:\n"
@@ -437,14 +444,23 @@ public class LinkageCheckerRuleTest {
       Assert.fail(
           "The rule should raise an EnforcerRuleException for artifacts missing dependencies");
     } catch (EnforcerRuleException ex) {
+
+      // Java 11 removed javax.activation package. Therefore the number of expected error differs
+      // between Java 8 and Java 11.
+      // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1856
+      int expectedErrorCount = System.getProperty("java.version").startsWith("1.8.") ? 112 : 117;
+
       // pass
-      verify(mockLog).error(ArgumentMatchers.startsWith("Linkage Checker rule found 112 errors:"));
+      verify(mockLog)
+          .error(
+              ArgumentMatchers.startsWith(
+                  "Linkage Checker rule found " + expectedErrorCount + " errors:"));
       assertEquals("Failed while checking class path. See above error report.", ex.getMessage());
     }
   }
 
   @Test
-  public void testExecute_shouldFilterExclusionRule()
+  public void testExecute_shouldFilterExclusionRule_java8()
       throws RepositoryException, URISyntaxException {
     try {
       // This artifact is known to contain classes missing dependencies
@@ -459,8 +475,15 @@ public class LinkageCheckerRuleTest {
           "The rule should raise an EnforcerRuleException for artifacts missing dependencies");
     } catch (EnforcerRuleException ex) {
       // pass.
-      // The number of errors was 112 in testExecute_shouldFailForBadProjectWithBundlePackaging
-      verify(mockLog).error(ArgumentMatchers.startsWith("Linkage Checker rule found 93 errors:"));
+
+      // Java 11 removed javax.activation package. Therefore the number of expected error differs
+      // between Java 8 and Java 11.
+      // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1856
+      int expectedErrorCount = System.getProperty("java.version").startsWith("1.8.") ? 93 : 98;
+      verify(mockLog)
+          .error(
+              ArgumentMatchers.startsWith(
+                  "Linkage Checker rule found " + expectedErrorCount + " errors:"));
       assertEquals("Failed while checking class path. See above error report.", ex.getMessage());
     }
   }

--- a/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
+++ b/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
@@ -235,7 +235,7 @@ public class LinkageCheckerRuleTest {
 
       String errorMessage = errorMessageCaptor.getValue();
 
-      // Java 11 removed javax.activation package. Therefore the number of expected error differs
+      // Java 11 removed javax.activation package. Therefore the number of expected errors differs
       // between Java 8 and Java 11.
       // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1856
       int expectedErrorCount = System.getProperty("java.version").startsWith("1.8.") ? 112 : 117;
@@ -445,7 +445,7 @@ public class LinkageCheckerRuleTest {
           "The rule should raise an EnforcerRuleException for artifacts missing dependencies");
     } catch (EnforcerRuleException ex) {
 
-      // Java 11 removed javax.activation package. Therefore the number of expected error differs
+      // Java 11 removed javax.activation package. Therefore the number of expected errors differs
       // between Java 8 and Java 11.
       // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1856
       int expectedErrorCount = System.getProperty("java.version").startsWith("1.8.") ? 112 : 117;
@@ -476,7 +476,7 @@ public class LinkageCheckerRuleTest {
     } catch (EnforcerRuleException ex) {
       // pass.
 
-      // Java 11 removed javax.activation package. Therefore the number of expected error differs
+      // Java 11 removed javax.activation package. Therefore the number of expected errors differs
       // between Java 8 and Java 11.
       // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1856
       int expectedErrorCount = System.getProperty("java.version").startsWith("1.8.") ? 93 : 98;

